packages/hoprd/src/admin.ts

```diff
@@ -32,53 +33,65 @@ export class AdminServer {
 
   constructor(private logs: LogStream, private host: string, private port: number) {}
 
-  async setup() {
-    let adminPath: string
-    for (const adminRelPath of ['../hopr-admin', './hopr-admin']) {
-      const adminPathInt = new URL(adminRelPath, import.meta.url).pathname
-      const nextPath = path.resolve(adminPathInt, '.next')
-      if (!fs.existsSync(nextPath)) {
-        continue
+  setup(): Promise<void> {
+    return new Promise(async (resolve, reject) => {
```

Comment:
Why rewriting using explicit `Promise` as a return value instead of `async` ?

Reply:
`HttpServer.listen()` uses a callback API and we want the method to *resolve* if, and only if, the `HttpServer` is listening and `reject` in all other cases such that we can handle the error at the caller